feat: added 'button' style param to task-card

jfmcquade opened the pull request:
PR Checklist

- [x] - Latest `master` branch merged
- [x] - PR title descriptive (can be used in release notes)

## Description

Adds a new style parameter option to the `task-card` component: `style: button`. Makes the task card display as a narrow button with the image appearing small and on the right. A toggle can also be provided in the child rows of the component. This is visually equivalent to the existing `button` component with `style: card`, but, as outlined in Issue #1608, the `task-card` component supports displaying the completion status of its related task, e.g. a green tick.

## Git Issues

Closes #1608 

## Screenshots/Videos

[comp_task_card](http://localhost:4200/template/comp_task_card) updated with examples of new style.

<img width="1000" alt="Screenshot 2022-11-01 at 12 52 22" src="https://user-images.githubusercontent.com/64838927/199236471-0eec4735-e902-4345-a3a6-00401a32497d.png">

<img width="376" alt="Screenshot 2022-11-01 at 12 52 55" src="https://user-images.githubusercontent.com/64838927/199236559-6ecc802e-8d18-4077-9aab-78c6c3ca72b5.png">

EsmeeIDEMS reviewed (commented):
I am a bit confused here - the `completed_field` parameter does not show up in the template `comp_task_card`, is it just that it's out of date? So how are the components in this template aware of whether they are completed?

jfmcquade commented:
> 



> I am a bit confused here - the `completed_field` parameter does not show up in the template `comp_task_card`, is it just that it's out of date? So how are the components in this template aware of whether they are completed?

You're right, the [comp_task_card](https://docs.google.com/spreadsheets/d/1_RY8HbsATOAZU2EDXY5R9OOE_YBEbW0EjTP5zxYP2aA/edit#gid=569531329) template was out of date. I've updated it, along with a minor tweak to the code to properly handle the "completed" status on cards for individual tasks.
I've also added styling for a task-card with `style: button` in the case where an image is not provided

EsmeeIDEMS reviewed (approved):
Functional test passed
